aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -429,29 +319,63 @@ module aptos_framework::stake {
         move_to(account, AptosCoinCapabilities { mint_cap })
     }
 
+    /// Initialize the validator account and give ownership to the signing account
+    /// except it leaves the ValidatorConfig to be set by another entity.
+    /// Note: this triggers setting the operator and owner, set it to the account's address
+    /// to set later.
+    public entry fun initialize_owner_only(account: &signer, operator: address, voter: address) acquires OwnerCapability, StakePool {
+        initialize_owner(account, false);
+
+        move_to(account, ValidatorConfig {
+            consensus_pubkey: vector::empty(),
+            network_addresses: vector::empty(),
+            fullnode_addresses: vector::empty(),
+            validator_index: 0,
+        });
+
+        let account_address = signer::address_of(account);
+        if (account_address != operator) {
+            set_operator(account, operator)
+        };
+        if (account_address != voter) {
+            set_delegated_voter(account, voter)
+        };
+    }
+
     /// Initialize the validator account and give ownership to the signing account.
-    public entry fun register_validator_candidate(
+    public entry fun initialize_validator(
         account: &signer,
         consensus_pubkey: vector<u8>,
         proof_of_possession: vector<u8>,
         network_addresses: vector<u8>,
         fullnode_addresses: vector<u8>,
     ) {
+        assert!(signature::bls12381_verify_proof_of_possession(consensus_pubkey, proof_of_possession), error::invalid_argument(EINVALID_PUBLIC_KEY));
+
+        initialize_owner(account, true);
```

Comment:
Why does this need to be done here in initialize_validator? Can't this be done by calling initialize_owner_only first if the owner and validator are the same?

Reply:
what's the comment about?